gsa/src/web/entities/bulkactions.js

```diff
@@ -229,4 +234,93 @@ BulkTagComponent.propTypes = {
   onClose: PropTypes.func.isRequired,
 };
 
-export default BulkTagComponent;
+export const useBulkExportEntities = () => {
+  const username = useUserName();
+  const userDefaultsSelector = useSelector(getUserSettingsDefaults);
+  const listExportFileName = userDefaultsSelector.getValueByName(
+    'listexportfilename',
+  );
+
+  const bulkExportEntities = useCallback(
+    ({
+      entities,
+      selected,
+      filter,
+      resourceType,
+      selectionType,
+      exportByFilterFunc,
+      exportByIdsFunc,
+      onDownload,
+      onError,
+    }) => {
+      if (selectionType === SelectionType.SELECTION_FILTER) {
+        const exportFilter = filter.all().toFilterString();
+
+        return exportByFilterFunc(exportFilter).then(response => {
+          const filename = generateFilename({
+            fileNameFormat: listExportFileName,
+            resourceType,
+            username,
+          });
+
+          const commandName =
+            'export' + capitalizeFirstLetter(resourceType) + 'ByFilter';
+
+          const xml = response?.data;
+          const exportedEntities = xml[commandName]?.exportedEntities;
```

Comment:
Could that ever happen that response has no data? Wouldn't that be an error that should be shown to the user? I really think the next line will crash in that case.

What about the exportedEntities if they aren't found? will this be handled in onDowload? Can that ever happen?